pkg/api/vlabs/validate.go

```diff
@@ -364,6 +364,23 @@ func (a *Properties) validateMasterProfile() error {
 	if m.SinglePlacementGroup != nil && m.AvailabilityProfile == AvailabilitySet {
 		return errors.New("singlePlacementGroup is only supported with VirtualMachineScaleSets")
 	}
+
+	// Check that distro has a valid value.
+	valid := false
```

Comment:
Nice add. Can you put this in a function and reuse it for the master profile and agent pool validations?

Reply:
done